[BUG]: automated sync lead to "collection not open"

### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

* Install a recent issue
* have an account connected to ankiweb
* enable automated sync
* close the app
* wait enough to trigger a sync on start (I think it's 10 minutes)
* open the app

### Expected behaviour

Sync, then a usable app. Or at least in case of error, just a forced restart and a working app.

### Actual behaviour

"Collection not opened" appear while the sync occur.

Sometime instead, I got a SQL error message before the Collection not opened.

Also, if I just press okay, I get the warning message every 20 seconds or so the the app is not responding, while the app actually works perfectly as long as I press on the "wait" button.

### Debug info

```text
AnkiDroid Version = 2.16alpha93-debug

Android Version = 13

Manufacturer = samsung

Model = SM-A226B

Hardware = mt6833

Webview User Agent = Mozilla/5.0 (Linux; Android 13; SM-A226B Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/110.0.5481.65 Mobile Safari/537.36

ACRA UUID = a2af6a9d-da31-4e59-bb5d-ff252cd2826d

New schema = false

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```


### (Optional) Anything else you want to share?

If I accept the android message to force quit the app and restart it it works as expected. If I stop the automated sync, it works correctly.

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

Yeah, after a third try, I got the exception

```
2023-03-20 23:19:41.965 25380-25380 CoroutineHelpersKt      com.ichi2.anki.debug                 E  java.lang.RuntimeException: android.database.sqlite.SQLiteException: error while compiling: "SELECT count() FROM (SELECT null FROM cards WHERE did = ? AND queue = 1 AND due < ? LIMIT ?)": CollectionNotOpen
                                                                                                    	at com.ichi2.libanki.sched.SchedV2._lrnForDeck(SchedV2.kt:1444)
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueList(SchedV2.kt:479)
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.kt:530)
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837)
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837)
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:97)
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:95)
                                                                                                    	at com.ichi2.anki.CollectionManager$withQueue$2.invokeSuspend(CollectionManager.kt:83)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
                                                                                                    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
                                                                                                    Caused by: android.database.sqlite.SQLiteException: error while compiling: "SELECT count() FROM (SELECT null FROM cards WHERE did = ? AND queue = 1 AND due < ? LIMIT ?)": CollectionNotOpen
                                                                                                    	at net.ankiweb.rsdroid.database.StreamingProtobufSQLiteCursor.<init>(StreamingProtobufSQLiteCursor.kt:259)
                                                                                                    	at net.ankiweb.rsdroid.database.RustSupportSQLiteDatabase.query(RustSupportSQLiteDatabase.kt:103)
                                                                                                    	at com.ichi2.utils.DatabaseChangeDecorator.query(Unknown Source:16)
                                                                                                    	at com.ichi2.libanki.DB.queryScalar(DB.kt:122)
                                                                                                    	at com.ichi2.libanki.sched.SchedV2._lrnForDeck(SchedV2.kt:1427)
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueList(SchedV2.kt:479) 
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.kt:530) 
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837) 
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:97) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:95) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withQueue$2.invokeSuspend(CollectionManager.kt:83) 
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) 
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) 
                                                                                                    	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42) 
                                                                                                    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664) 
                                                                                                    Caused by: net.ankiweb.rsdroid.exceptions.BackendInvalidInputException$BackendCollectionNotOpenException: CollectionNotOpen
                                                                                                    	at net.ankiweb.rsdroid.exceptions.BackendInvalidInputException$Companion.fromInvalidInputError(BackendInvalidInputException.kt:35)
                                                                                                    	at net.ankiweb.rsdroid.BackendException$Companion.fromError(BackendException.kt:114)
                                                                                                    	at net.ankiweb.rsdroid.BackendKt.unpackResult(Backend.kt:323)
                                                                                                    	at net.ankiweb.rsdroid.BackendKt.access$unpackResult(Backend.kt:1)
                                                                                                    	at net.ankiweb.rsdroid.Backend$runMethodRaw$1.invoke(Backend.kt:130)
                                                                                                    	at net.ankiweb.rsdroid.Backend$runMethodRaw$1.invoke(Backend.kt:129)
                                                                                                    	at net.ankiweb.rsdroid.Backend.withBackend(Backend.kt:157)
                                                                                                    	at net.ankiweb.rsdroid.Backend.runMethodRaw(Backend.kt:129)
                                                                                                    	at anki.backend.GeneratedBackend.runDbCommandProtoRaw(GeneratedBackend.kt:1699)
                                                                                                    	at anki.backend.GeneratedBackend.runDbCommandProto(GeneratedBackend.kt:1709)
                                                                                                    	at net.ankiweb.rsdroid.Backend.fullQueryProto(Backend.kt:223)
                                                                                                    	at net.ankiweb.rsdroid.database.Session.fullQueryProto(Session.kt:79)
                                                                                                    	at net.ankiweb.rsdroid.database.StreamingProtobufSQLiteCursor.<init>(StreamingProtobufSQLiteCursor.kt:255)
                                                                                                    	at net.ankiweb.rsdroid.database.RustSupportSQLiteDatabase.query(RustSupportSQLiteDatabase.kt:103) 
                                                                                                    	at com.ichi2.utils.DatabaseChangeDecorator.query(Unknown Source:16) 
                                                                                                    	at com.ichi2.libanki.DB.queryScalar(DB.kt:122) 
                                                                                                    	at com.ichi2.libanki.sched.SchedV2._lrnForDeck(SchedV2.kt:1427) 
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueList(SchedV2.kt:479) 
                                                                                                    	at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.kt:530) 
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837) 
                                                                                                    	at com.ichi2.anki.DeckPicker$updateDeckList$2$1$deckData$1.invoke(DeckPicker.kt:1837) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:97) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withCol$2.invoke(CollectionManager.kt:95) 
                                                                                                    	at com.ichi2.anki.CollectionManager$withQueue$2.invokeSuspend(CollectionManager.kt:83) 
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) 
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) 
                                                                                                    	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42) 
                                                                                                    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677) 
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664) 
```

--- a/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/libanki/sched/SchedV2.kt
@@ -461,7 +461,7 @@ open class SchedV2(col: Collection) : AbstractSched(col) {
             // new
             var nlim = _deckNewLimitSingle(deck, false)
             var plim: Int? = null
-            if (!p.isNullOrEmpty()) {
+            if (p?.isNotEmpty() == true) {
                 val parentLims = lims[Decks.normalizeName(p)]
                 // 'temporary for diagnosis of bug #6383'
                 Assert.that(

